src/Catalyst.Core.Modules.Web3/Controllers/EthController.cs

```diff
@@ -21,102 +21,45 @@
-        }
-
-        [HttpGet]
-        public ResultWrapper<byte[]> eth_getStorageAt(Address address,
-            UInt256 positionIndex,
-            BlockParameter blockParameter)
-        {
-            return _ethRpcService.eth_getStorageAt(address, positionIndex, blockParameter);
-        }
-
-        [HttpGet]
-        public UInt256? eth_getTransactionCount(Address address, BlockParameter blockParameter)
-        {
-            return _ethRpcService.eth_getTransactionCount(address, blockParameter);
-        }
-
-        [HttpGet]
-        public ResultWrapper<byte[]> eth_getCode(Address address, BlockParameter blockParameter)
-        {
-            return _ethRpcService.eth_getCode(address, blockParameter);
-        }
-
-        [HttpGet]
-        public Keccak eth_sendRawTransaction(byte[] transaction)
-        {
-            return _ethRpcService.eth_sendRawTransaction(transaction);
-        }
-
-        [HttpGet]
-        public ResultWrapper<byte[]> eth_call(TransactionForRpc transactionCall, BlockParameter blockParameter = null)
-        {
-            return _ethRpcService.eth_call(transactionCall, blockParameter);
-        }
-
-        [HttpGet]
-        public UInt256? eth_estimateGas(TransactionForRpc transactionCall)
-        {
-            return _ethRpcService.eth_estimateGas(transactionCall);
-        }
-
-        [HttpGet]
-        public BlockForRpc eth_getBlockByNumber(BlockParameter blockParameter, bool returnFullTransactionObjects)
-        {
-            return _ethRpcService.eth_getBlockByNumber(blockParameter, returnFullTransactionObjects);
-        }
-
-        [HttpGet]
-        public TransactionForRpc eth_getTransactionByHash(Keccak transactionHash)
-        {
-            return _ethRpcService.eth_getTransactionByHash(transactionHash);
-        }
-
-        [HttpGet]
-        public TransactionForRpc eth_getTransactionByBlockNumberAndIndex(BlockParameter blockParameter,
-            UInt256 positionIndex)
-        {
-            return _ethRpcService.eth_getTransactionByBlockNumberAndIndex(blockParameter, positionIndex);
-        }
-
-        [HttpGet]
-        public ReceiptForRpc eth_getTransactionReceipt(Keccak txHashData)
+        public EthController(IWeb3EthApi web3EthApi, IWeb3HandlerResolver handlerResolver, IJsonSerializer jsonSerializer)
         {
-            return _ethRpcService.eth_getTransactionReceipt(txHashData);
+            _web3EthApi = web3EthApi ?? throw new ArgumentNullException(nameof(web3EthApi));
+            _handlerResolver = handlerResolver ?? throw new ArgumentNullException(nameof(handlerResolver));
+            _jsonSerializer = jsonSerializer ?? throw new ArgumentNullException(nameof(jsonSerializer));
         }
 
-        [HttpGet]
-        public BlockForRpc eth_getUncleByBlockHashAndIndex(Keccak blockHashData, UInt256 positionIndex)
+        [HttpPost]
+        public JsonRpcResponse Request([FromBody] JsonRpcRequest request)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename the parameter 'request' so that it does not duplicate the method name.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)